src/test/java/StringManipulationTest.java

```diff
@@ -1,149 +1,200 @@
+        assertEquals(restoreString, "rat"); //asserting restored string
     }
 
     @Test
-    public void testRestoreString2()
-    {
-        fail("Not yet implemented");
-
+    public void testRestoreString2() {
+        assertThrows(IllegalArgumentException.class, () -> {
+            manipulatedString.setString("invalid"); //setting string
+            int[] array = {1, 0}; //invalid indices array
+            manipulatedString.restoreString(array); //should throw exception
+        });
     }
 
     @Test
-    public void testRestoreString3()
-    {
-        fail("Not yet implemented");
-
+    public void testRestoreString3() {
+        assertThrows(IllegalArgumentException.class, () -> {
+            manipulatedString.setString("outofbounds"); //setting string
+            int[] array = {1, 0, 20}; //out of bounds indices array
+            manipulatedString.restoreString(array); //should throw exception
+        });
     }
 
     @Test
-    public void testRestoreString4()
-    {
-        fail("Not yet implemented");
-
+    public void testRestoreString4() {
+        // Test case to verify the restoreString() method when the input string is "team"
+        manipulatedString.setString("team");
+        // and the indices array is {-1, 0, 1, 2}.
+        int[] array = new int[]{-1, 0, 1, 2};
+        // It should throw an IndexOutOfBoundsException.
+        assertThrows(IndexOutOfBoundsException.class, ()->{manipulatedString.restoreString(array);});
     }
 
     @Test
-    public void testRestoreString5()
-    {
-        fail("Not yet implemented");
+    public void testRestoreString5(){
+        // Test case to verify the restoreString() method when the input string is "group"
+        manipulatedString.setString("group");
+        // and the indices array is {5, 0, 1, 2, 3}.
+        int[] array = new int[]{5, 0, 1, 2, 3};
+        // It should throw an IndexOutOfBoundsException.
+        assertThrows(IndexOutOfBoundsException.class, ()->{manipulatedString.restoreString(array);});
+    }
 
+    @Test
+    public void testRestoreString6() {
+        // Test case to verify the restoreString() method when the input string is null
+        manipulatedString.setString(null);
+        // and the indices array is {2, 3, 5, 4, 1, 0}.
+        int[] array = new int[]{2, 3, 5, 4, 1, 0};
+        // It should throw an IllegalArgumentException.
+        assertThrows(IllegalArgumentException.class, ()->{manipulatedString.restoreString(array);});
     }
 
-}
+    @Test
+    public void testRestoreString7() {
+        // Test case to verify the restoreString() method when the input string is "MASTER"
+        manipulatedString.setString("MASTER");
+        // and the indices array is {5, 4, 0, 1, 2, 3}.
+        int[] array = new int[]{5, 4, 0, 1, 2, 3};
+        // It should return the expected output "STERAM".
+        assertEquals("STERAM", manipulatedString.restoreString(array));
+    }
```

Comment:
My only suggestion here would be to adjust this test to include whitespaces and move those whitespaces around along with the characters. Including this kind of test would avoid any doubt on whether the method could handle strings with numerous whitespaces in awkward indexes like the beginning and end. Whitespaces are treated like any other character, but testing for these along with other punctuation and character symbols would make these tests highly comprehensive.

Reply:
This has been updated. Thank you for the feedback.